File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <bits/stdc++.h>
using namespace std;
typedef queue<int> qi;
typedef vector<int> vi;
const int MAX = 1005;

bool visited[MAX];
vi graph[MAX];

void dfs(int v)
{
    visited[v] = true;
    cout << v << ' ';
    for (int i : graph[v])
    {
        if (!visited[i])
            dfs(i);
    }
}

void bfs(int v)
{
    qi queue = qi();
    queue.push(v);
    visited[v] = true;
    
    while (!queue.empty())
    {
        int node = queue.front();
        queue.pop();
        cout << node << ' ';
        for (int i : graph[node])
        {
            if (!visited[i])
            {
                visited[i] = true;
                queue.push(i);
            }
        }
    }
}

int n, m, v = 0;
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0); cout.tie(0);

    memset(visited, false, sizeof visited);
    cin >> n >> m >> v;
    for (int i = 0; i < m; i++)
    {
        int st, en = 0;
        cin >> st >> en;
        graph[st].push_back(en);
        graph[en].push_back(st);
    }
    for (int i = 1; i <= n; i++)
        sort(graph[i].begin(), graph[i].end());
    
    dfs(v); cout << '\n';
    memset(visited, false, sizeof visited);
    bfs(v); cout << '\n';
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

# 첫 번째 줄에서 n, m, v 입력 받기
n, m, v = map(int, input().split())

# 그래프 초기화
graph = [[] for _ in range(n + 1)]

# 이후 m개의 간선 정보 입력 받기
for _ in range(m):
    U, V = map(int, input().split())
    graph[U].append(V)
    graph[V].append(U)

# 그래프의 각 리스트 정렬
for connections in graph:
    connections.sort()

def dfs(v, visited):
    print(v, end=' ')
    visited[v] = True
    for neighbor in graph[v]:
        if not visited[neighbor]:
            dfs(neighbor, visited)

def bfs(v, visited):
    queue = deque([v])
    visited[v] = True
    while queue:
        current = queue.popleft()
        print(current, end=' ')
        for neighbor in graph[current]:
            if not visited[neighbor]:
                queue.append(neighbor)
                visited[neighbor] = True

# 방문 확인용 배열 초기화
visited_dfs = [False] * (n + 1)
visited_bfs = [False] * (n + 1)

# DFS와 BFS 실행
dfs(v, visited_dfs)
print()
bfs(v, visited_bfs)